src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs

```diff
@@ -108,6 +108,15 @@ private void ProcessType(INamedTypeSymbol typeSymbol)
 
 				if (isDependencyObject && typeSymbol.TypeKind == TypeKind.Class)
 				{
+					if (typeSymbol.Is(_iosViewSymbol))
+					{
+						throw new InvalidOperationException("A 'UIKit.UIView' shouldn't implement 'DependencyObject'. Inherit 'FrameworkElement' instead.");
+					}
+					else if (typeSymbol.Is(_androidViewSymbol))
+					{
+						throw new InvalidOperationException("An 'Android.Views.View' shouldn't implement 'DependencyObject'. Inherit 'FrameworkElement' instead.");
+					}
```

Comment:
Should we have something similar for macOS `NSView`?

Reply:
Yes, `NSView` is having the same issue.